### PR TITLE
Return semantic error in octopipe

### PR DESCRIPTION
**Pull request by thallesfreitaszup**
## Issue 
Octopipe was not returning a semantic error when a namespace wasn't found because an error was not verified and returned and because of this, that error was overrided
## Solution
Added a if for that error and log  and return that error in response

### CODE REVIEW

File: octopipe/pkg/execution/manager.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type ManagerUseCases interface {
 		executionID *primitive.ObjectID, step *pipeline.Step,
 	) (*primitive.ObjectID, error)
 	ExecutionError(executionID *primitive.ObjectID, pipelineError error) error
-	ExecutionFinished(executionID *primitive.ObjectID, pipelineError error) error
+	ExecutionFinished(executionID *primitive.ObjectID, ṕipelineError chan error ) error
```
Comment (lucasbfernandes) [inline octopipe/pkg/execution/manager.go:34]: Remove space after error

File: octopipe/pkg/mozart/mozart.go [hunk 2]
```diff
@@ -52,11 +52,14 @@ func NewMozart(mozartManager *MozartManager, deployment *deployment.Deployment)
 }
 
 // TODO: Remove deployment param
-func (mozart *Mozart) Do(deployment *deployment.Deployment) {
-	go mozart.asyncStartPipeline(deployment)
+func (mozart *Mozart) Do(deployment *deployment.Deployment) string {
+	errors := make(chan error, 1)
```
Comment (lucasbfernandes) [inline octopipe/pkg/mozart/mozart.go:56]: Why are you using 1?

File: octopipe/pkg/mozart/mozart.go [hunk 3]
```diff
@@ -52,11 +52,14 @@ func NewMozart(mozartManager *MozartManager, deployment *deployment.Deployment)
...
+	go mozart.asyncStartPipeline(deployment, errors)
+	errObject := <- errors
+	return errObject.Error()
 }
 
-func (mozart *Mozart) asyncStartPipeline(deployment *deployment.Deployment) {
+func (mozart *Mozart) asyncStartPipeline(deployment *deployment.Deployment,errors chan error)  {
```
Comment (lucasbfernandes) [inline octopipe/pkg/mozart/mozart.go:62]: Space after comma

File: octopipe/pkg/api/pipeline.go [hunk 4]
```diff
@@ -38,7 +38,11 @@ func (api *PipelineAPI) startPipeline(ctx *gin.Context) {
 	var deployment *deployment.Deployment
 	ctx.Bind(&deployment)
 
-	api.mozart.Start(deployment)
+	err := api.mozart.Start(deployment)
 
-	ctx.JSON(201, nil)
+	if err != "" {
```
[the remark this answers is not in the document]
Comment (thallesfreitaszup) [reply octopipe/pkg/api/pipeline.go:43]: https://stackoverflow.com/questions/53859961/cannot-convert-nil-to-type-string

File: octopipe/pkg/execution/manager.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ type ManagerUseCases interface {
 		executionID *primitive.ObjectID, step *pipeline.Step,
 	) (*primitive.ObjectID, error)
 	ExecutionError(executionID *primitive.ObjectID, pipelineError error) error
-	ExecutionFinished(executionID *primitive.ObjectID, pipelineError error) error
+	ExecutionFinished(executionID *primitive.ObjectID, ṕipelineError chan error) error
```
Comment (cpgo) [inline octopipe/pkg/execution/manager.go:34]: `ṕipelineError`
Strange character here

**Pull request closed without merging**